### ISSUE TITLE
0.5.0-beta readiness tracker

### ISSUE THREAD

**Issue by MikeGoldsmith** [open-telemetry/opentelemetry-dotnet#506]
This is a tracking issue for the requirements outlined in the beta release document: https://docs.google.com/document/d/1eviggoIguOS89dgKL-8ntUOCfPP6FLWAoQkGoBDsld0/edit#

## Traces
- [x] OpenTelemetry SpanWrapper on top of .NET Activity.
- [x] Stable API of every spec'd feature #573
- [x] SDK Implementation of every spec'd feature
- [x] #574
- [x] #575
- [x] #576 
- [x] #577
- [x] #578
- [x] Basic tracing
- [x] Named tracers
- [ ] Tracers use OTEP 66 Context mechanism - This is an implementation detail and does not affect the API
## Metrics
- [ ] Documentation
- [x] Examples
- [x] Stable API of every spec'd feature
- [x] SDK Implementation of every spec'd feature - No spec exists, so we have own spec.
- [x] Basic SDK
- [x] Add observer open-telemetry/opentelemetry-specification#430
- [x] Remove gauge open-telemetry/opentelemetry-specification#430

## Resources
- [x] Stable API of every spec'd feature
- [ ] SDK Implementation of every spec'd feature
## Context Propagation
- [ ] Stable API of every spec'd feature
- [x] Basic context mechanism
- [ ] Global Propagators API #581
- [ ] Global Context API (get and set active context) #461
- [ ] SDK Implementation of every spec'd feature
- [ ] Composite Propagator #582
- [x] Basic context mechanism
## Exporters
- [x] OpenTelemetry Native - Collector Exporter
- [x] Prometheus
- [x] Jaeger
- [x] Zipkin
## Instrumentation
- [x] HTTP
- [x] SQL
## Docs and examples
- [x] Instrumenting using Activity API or Span Wrapper for
- [x] Exporting telemetry
- [x] Configuring pipeline
- [ ] Adding instrumentation (aka adapters or auto-collectors)

**Comment by hacst** [comment]
@MikeGoldsmith Any updates on this or ETA for the beta? [The general beta announcement post](https://medium.com/opentelemetry/opentelemetry-is-officially-in-beta-352fa859db10) said "The .Net API + SDK will follow shortly." but it is really hard to get a feeling for how close this actually is as an outsider. Thanks.

**Comment by MikeGoldsmith** [comment]
Hi @hacst 

I'll be updating this today / over the weekend ready for our next SIG call on Tuesday. With this complete, we should have a better picture of what is left for us to complete before we can release our first Beta release.

**Comment by giuliano-barberi-tf** [comment]
@MikeGoldsmith @SergeyKanzhelev I'm wondering if there's been any updates to the beta readiness. As an outsider I've also read a few blog posts about using OpenTelemetry in .NET but in this repo I couldn't find whether the beta is close or not. It's been a few months since the last update I'm wondering if anything has changed. Thanks!

**Comment by cijothomas** [comment]
@giuliano-barberi-tf We are currently focused on .NET Activity API integration with OpenTelemetry. We understand a lot of people are waiting for beta to arrive - I'd tentatively say June end based on this https://github.com/open-telemetry/opentelemetry-dotnet/issues/684

I'll put on an agenda on next weeks SIG meeting, to discuss more on beta release, and update back here.

**Comment by czs1993** [comment]
The beta is completed? the readliness haven’t update for a long time. Thanks!

**Comment by cijothomas** [comment]
> The beta is completed? the readliness haven’t update for a long time. Thanks!

No. This repo has never released a beta. We expect to have discussion on this tomorrow's SIG meeting.

**Comment by czs1993** [comment]
@cijothomas  thanks, looking forward to the repo update.

**Comment by cijothomas** [comment]
Updated: As discussed, beta date is postponed from July 17 to July 24, so as to have DiagnosticSourcepreview7 available on Nuget. (OT packages depends on this package)

**Comment by cijothomas** [comment]
Since we are now able to track items using milestones, we no longer need this. closing. Let us know if this is still needed.
To know whats coming in next beta, this should help: https://github.com/open-telemetry/opentelemetry-dotnet/milestone/8
